src/sentry/integrations/slack/unfurl/discover.py

```diff
@@ -0,0 +1,110 @@
+import logging
+import re
+from typing import Any, List, Mapping
+from urllib.parse import urlparse
+
+from django.http.request import QueryDict
+
+from sentry import features
+from sentry.api import client
+from sentry.charts import generate_chart
+from sentry.charts.types import ChartType
+from sentry.integrations.slack.utils import logger
+from sentry.models import ApiKey
+
+from . import Handler, UnfurlableUrl, UnfurledUrl
+
+# The display modes on the frontend are defined in app/utils/discover/types.tsx
+display_modes: Mapping[str, ChartType] = {
+    "default": ChartType.SLACK_DISCOVER_TOTAL_PERIOD,
+    "daily": ChartType.SLACK_DISCOVER_TOTAL_DAILY,
+    "top5": ChartType.SLACK_DISCOVER_TOP5_PERIOD,
+    "dailytop5": ChartType.SLACK_DISCOVER_TOP5_DAILY,
+    # TODO(epurkhiser): Previous period
+}
+
+TOP_N = 5
+
+
+def unfurl_discover(data, integration, links: List[UnfurlableUrl]) -> UnfurledUrl:
+    orgs_by_slug = {org.slug: org for org in integration.organizations.all()}
+    unfurls = {}
+
+    for link in links:
+        org_slug = link.args["org_slug"]
+        org = orgs_by_slug.get(org_slug)
+
+        # If the link shared is an org w/o the slack integration do not unfurl
+        if not org:
+            continue
+        if not features.has("organizations:chart-unfurls", org):
+            continue
+
+        params = link.args["query"]
+        params.setlist("order", params.get("sort"))
+        display_mode = str(link.args["query"].get("display", "default"))
+
+        if "daily" in display_mode:
+            params.setlist("interval", ["1d"])
+        if "top5" in display_mode:
+            params.setlist("topEvents", [f"{TOP_N}"])
+
+        logging.info(params)
```

Comment:
@EvanPurkhiser why not make a local logger here so we know what file this is coming from?

Reply:
oh this is leftover debugging :)